src/transformers/tokenization_tapas.py

```diff
@@ -323,6 +326,829 @@ def save_vocabulary(self, save_directory: str, filename_prefix: Optional[str] =
+                    "The PreTrainedTokenizerBase.prepare_for_model `return_lengths` parameter is deprecated. "
+                    "Please use `return_length` instead.",
+                    FutureWarning,
+                )
+            return_length = kwargs["return_lengths"]
+
+        return self._batch_encode_plus(
+            table=table,
+            queries=queries,
+            answer_coordinates=answer_coordinates,
+            answer_texts=answer_texts,
+            add_special_tokens=add_special_tokens,
+            padding_strategy=padding_strategy,
+            truncation_strategy=truncation_strategy,
+            max_length=max_length,
+            stride=stride,
+            is_split_into_words=is_split_into_words,
+            pad_to_multiple_of=pad_to_multiple_of,
+            return_tensors=return_tensors,
+            return_token_type_ids=return_token_type_ids,
+            return_attention_mask=return_attention_mask,
+            return_overflowing_tokens=return_overflowing_tokens,
+            return_special_tokens_mask=return_special_tokens_mask,
+            return_offsets_mapping=return_offsets_mapping,
+            return_length=return_length,
+            verbose=verbose,
+            **kwargs,
+        )
+
+    def _batch_encode_plus(
+        self,
+        table,
+        queries: Union[
+            List[TextInput],
+            List[PreTokenizedInput],
+            List[EncodedInput],
+        ],
+        answer_coordinates: Optional[List[Tuple]] = None,
+        answer_texts: Optional[List[TextInput]] = None,
+        add_special_tokens: bool = True,
+        padding_strategy: PaddingStrategy = PaddingStrategy.DO_NOT_PAD,
+        truncation_strategy: TruncationStrategy = TruncationStrategy.DO_NOT_TRUNCATE,
+        max_length: Optional[int] = None,
+        stride: int = 0,
+        is_split_into_words: bool = False,
+        pad_to_multiple_of: Optional[int] = None,
+        return_tensors: Optional[Union[str, TensorType]] = None,
+        return_token_type_ids: Optional[bool] = True,
+        return_attention_mask: Optional[bool] = None,
+        return_overflowing_tokens: bool = False,
+        return_special_tokens_mask: bool = False,
+        return_offsets_mapping: bool = False,
+        return_length: bool = False,
+        verbose: bool = True,
+        **kwargs
+    ) -> BatchEncoding:
+
+        table_tokens = self._tokenize_table(table)
+
+        queries_tokens = []
+        queries_ids = []
+        for query in queries:
+            query_tokens = self.tokenize(query)
+            queries_tokens.append(query_tokens)
+            queries_ids.append(self.convert_tokens_to_ids(query_tokens))
+
+        num_rows = self._get_num_rows(table, self.drop_rows_to_fit)
+        num_columns = self._get_num_columns(table)
+
+        _, _, num_tokens = self._get_table_boundaries(table_tokens)
+
+        table_data = list(self._get_table_values(table_tokens, num_columns, num_rows, num_tokens))
```

Comment:
You're correct, however, as noted in the description, I think the previous truncation should still be implemented, as we want to be as close as possible to the original API. I didn't reimplement it here because I wanted to do it fast, but I should work on re-implementing that ASAP. 

This should leverage all the methods/attributes you mentioned.